workspaces/topology/plugins/topology/tests/utils/topologyHelper.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import AxeBuilder from '@axe-core/playwright';
+import { expect, TestInfo, type Locator, type Page } from '@playwright/test';
+import { TopologyMessages, templateToPattern } from './translations';
+
+export class Common {
+  page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async verifyHeading(heading: string) {
+    const headingLocator = this.page
+      .locator('h1, h2, h3, h4, h5, h6')
+      .filter({ hasText: heading })
+      .first();
+    await headingLocator.waitFor({ state: 'visible', timeout: 30000 });
+    await expect(headingLocator).toBeVisible();
+  }
+
+  async clickButton(
+    label: string,
+    clickOpts?: Parameters<Locator['click']>[0],
+    getByTextOpts: Parameters<Locator['getByText']>[1] = { exact: true },
+  ) {
+    const muiButtonLabel = 'span[class^="MuiButton-label"]';
+    const selector = `${muiButtonLabel}:has-text("${label}")`;
+    const button = this.page
+      .locator(selector)
+      .getByText(label, getByTextOpts)
+      .first();
+    await button.waitFor({ state: 'visible' });
+    await button.click(clickOpts);
+  }
```

Comment:
can we stop using these functions? they are needlessly complicated for what they do